hive/README_zh_CN.md

```diff
@@ -0,0 +1,208 @@
+cd TiBigData
+# 在编译之前你需要先编译 TiKV 的 java 客户端
+./.ci/build-client-java.sh
+# 编译 Hive Storage Handler，以 hive-2.2.0 为例
+mvn clean package -DskipTests -am -pl hive/hive-2.2.0
+```
+因为 Hive 与 Hadoop 的依赖较多，根据网络状况与电脑配置，整个过程可能需要花费 10 到 30 分钟，国内用户推荐使用国内 maven 仓库来加速。
+
+## 3 安装 Hive Storage Handler
+
+Hive 使用第三方的 Storage Handler 需要添加依赖。这里我们推荐两种方式添加 TiDB Hive Storage Handler 的 jar 包。
+
+### 3.1 利用 ADD JAR 语法
+
+这种做法的好处是不会给 Hive 引入额外的依赖，但是每次启动 HiveCli 或者连接到 HiveServer2 的时候都需要跑一次 ADD JAR 语法。
+
+```bash
+# 将编译出来的 jar 包上传到 hdfs 上
+hadoop fs -put $TIBIGDATA_HOME/hive/hive-2.2.0/target/hive-tidb-storage-handler-2.2.0-0.0.5-SNAPSHOT.jar /tmp
+# 启动 Hive 客户端或者连接至 HiveServer2 后执行如下命令初始化
+ADD JAR hdfs:///tmp/hive-tidb-storage-handler-2.2.0-0.0.5-SNAPSHOT.jar;
+```
+
+### 3.2 直接将依赖放入 Hive 的 lib 目录下
+
+```bash
+cp $TIBIGDATA_HOME/hive/hive-2.2.0/target/hive-tidb-storage-handler-2.2.0-0.0.5-SNAPSHOT.jar $HIVE_HOME/lib/
+```
+
+## 4 利用 Hive 读写 TiDB
+我们尝试在 TiDB 内创建一张表。
+
+```bash
+# 连接至 TiDB
+mysql --host 127.0.0.1 --port 4000 -uroot --database test
+```
+
+当前**不支持**在 Hive 内向 TiDB 写数据，所以我们在 TiDB 建表并插入一条数据：
+
+```sql
+-- 这段 sql 跑在 TiDB 内
+CREATE TABLE `people`(
+  `id` int,
+  `name` varchar(16)
+);
+
+INSERT INTO `people`(`id`,`name`) VALUES (1,'zs');
+```
+
+在 Hive 建 TiDB 的映射表，这里的字段随便写，TiBigData 会自动修正字段：
+
+```sql
+-- 如果没有将依赖拷贝进 Hive 的 lib 目录，需要使用 ADD JAR 先添加依赖
+-- ADD JAR hdfs:///tmp/hive-tidb-storage-handler-2.2.0-0.0.5-SNAPSHOT.jar;
+CREATE TABLE `people`(
+ `fake_column` int
+)
+STORED BY
+  'io.tidb.bigdata.hive.TiDBStorageHandler'
+TBLPROPERTIES (
+  'tidb.database.url'='jdbc:mysql://localhost:4000/',
+  'tidb.database.name'='test',
+  'tidb.table.name'='people',
+  'tidb.username'='root',
+  'tidb.password'=''
+ );
+```
+
+在 Hive 内查看表结构：
+
+```sql
+SHOW CREATE TABLE `people`;
+```
+
+可以看到表的字段已经被自动修正了：
+
+```sql
+hive (default)> SHOW CREATE TABLE `people`;
+OK
+createtab_stmt
+CREATE TABLE `people`(
+  `id` int COMMENT 'from deserializer',
+  `name` string COMMENT 'from deserializer')
+ROW FORMAT SERDE
+  'io.tidb.bigdata.hive.TiDBSerde'
+STORED BY
+  'io.tidb.bigdata.hive.TiDBStorageHandler'
+WITH SERDEPROPERTIES (
+  'serialization.format'='1')
+LOCATION
+  'hdfs://XXX:8020/user/hive/warehouse/people'
+TBLPROPERTIES (
+  'COLUMN_STATS_ACCURATE'='{\"BASIC_STATS\":\"true\"}',
+  'numFiles'='0',
+  'numRows'='0',
+  'rawDataSize'='0',
+  'tidb.database.name'='test',
+  'tidb.database.url'='jdbc:mysql://localhost:4000/',
+  'tidb.password'='',
+  'tidb.table.name'='people',
+  'tidb.username'='root',
+  'totalSize'='0',
+  'transient_lastDdlTime'='1642779466')
+Time taken: 2.276 seconds, Fetched: 23 row(s)
+hive (default)>
+```
+
+在 Hive 内查询 TiDB 表：
+
+```sql
+SELECT * FROM `people`;
+```
+
+你会得到以下结果：
+
+```sql
+hive (default)> SELECT * FROM `people`;
+OK
+people.id	people.name
+1	zs
+Time taken: 4.979 seconds, Fetched: 1 row(s)
+hive (default)>
+```
+
+至此，你已经知道如何在 Hive 内使用 TiBigData 了。更多高级的功能以及配置调优可参考下面的章节。
+
+## 5 Flink 与 TiDB 的类型映射
```

Comment:
I guess we need to have a different type mapping table for hive?

Reply:
At this time, the schema of tidb is not stored in hive-metastore and hive will query it in real time. This can effectively solve the column change problem caused by DDL.